compiler/noirc_frontend/src/parser/parser/primitives.rs

```diff
@@ -32,6 +33,16 @@ pub(super) fn token_kind(token_kind: TokenKind) -> impl NoirParser<Token> {
     })
 }
 
+pub(super) fn path_segment() -> impl NoirParser<PathSegment> {
+    ident()
+        .then(turbofish(super::parse_type()))
```

Comment:
Hmm, yeah I  could see this as potentially problematic since types themselves can contain path_segments (parsing struct types) so this could lead to infinite recursion.